Data Provider

akoushke opened the pull request.

lalli-flores commented on src/components/hooks/useAdapter.js:
```diff
@@ -0,0 +1,12 @@
+import {AdapterContext} from '../../components/';
+
+/**
+ * Custom hook that returns shared context adapter data across components.
```
```suggestion
 * Custom hook that returns shared data adapter context across components.
```

lalli-flores commented on src/components/WebexActivity/WebexActivity.js:
```diff
@@ -71,8 +72,4 @@ export default function WebexActivity(props) {
 
 WebexActivity.propTypes = {
   activityID: PropTypes.string.isRequired,
```
Remove trailing comma
```suggestion
  activityID: PropTypes.string.isRequired
```

akoushke replied:
you cant. prettify requires it.

lalli-flores replied:
I see why this is off, we shouldn't be passing an adapter to `Header` anymore, it need it for the `WebexAvatar`, but `WebexAvatar` has its own adapter now.

lalli-flores commented on src/components/WebexActivity/WebexActivity.stories.js:
```diff
@@ -2,56 +2,84 @@ import React from 'react';
-import {activities, people} from '../../data';
-
-import WebexActivity from './WebexActivity';
+import * as webexData from '../../data';
```
If you make it a default export you won't need to use `*`.
Also, can you rename this to `testData` or `jsonData` as to not confuse it with the adapter provider?

lalli-flores commented on src/components/WebexActivityStream/README.md:
````diff
@@ -19,13 +19,19 @@ To see all the different possible states of the Webex Activity Stream component,
 
    ```js
-   const jsonAdapter = new RoomsJSONAdapter(rooms);
+   const jsonAdapter = {
````
This should be creating a `WebexJSONAdapter`

lalli-flores commented on src/components/WebexActivityStream/WebexActivityStream.js:
```diff
@@ -157,9 +158,4 @@ export default function WebexActivityStream(props) {
 
 WebexActivityStream.propTypes = {
   roomID: PropTypes.string.isRequired,
```
Remove trailing comma, please
```suggestion
  roomID: PropTypes.string.isRequired
```

lalli-flores commented on src/components/WebexAvatar/README.md:
````diff
@@ -19,13 +19,15 @@ To see all the different possible states of the Webex Avatar component, you can
 
    ```js
-   const jsonAdapter = new PeopleJSONAdapter(people);
+   const jsonAdapter = {peopleAdapter: new PeopleJSONAdapter(people)};
````
This should be creating a `WebexJSONAdapter`

lalli-flores commented on src/components/WebexAvatar/WebexAvatar.js:
```diff
@@ -2,16 +2,16 @@ import React from 'react';
 }
 
 WebexAvatar.propTypes = {
   personID: PropTypes.string.isRequired,
```
Remove trailing comma, please
```suggestion
  personID: PropTypes.string.isRequired
```

lalli-flores commented on src/components/WebexData/README.md:
```diff
@@ -0,0 +1,3 @@
+# Webex Activity Data Provider Component
+
+LALLI HELP ME HERE PLEASE. MY BRAIN IS FRIED RIGHT NOW!
```
Do you want to me to open a PR for this after this is merged?

akoushke replied:
if you want

lalli-flores commented on src/components/WebexData/WebexData.js:
```diff
@@ -0,0 +1,17 @@
+
+export const AdapterContext = React.createContext();
+
+export default function WebexData(props) {
```
Since this is a new component, use the spread operator for the props like we agreed on Adam's PR.
```suggestion
export default function WebexData({adapter, children}) {
```

lalli-flores commented on src/components/WebexData/WebexData.js:
```diff
@@ -0,0 +1,17 @@
+
+export const AdapterContext = React.createContext();
+
+export default function WebexData(props) {
```
I think the team was leaning towards renaming this `WebexDataProvider`

akoushke replied:
yeah

lalli-flores commented on src/components/WebexData/WebexData.js:
```diff
@@ -0,0 +1,17 @@
+    activitiesAdapter: PropTypes.object.isRequired,
+    peopleAdapter: PropTypes.object.isRequired,
+  }).isRequired,
+  children: PropTypes.node.isRequired,
```
No trailing comma, please
```suggestion
  children: PropTypes.node.isRequired
```

akoushke replied:
see my other comments

lalli-flores commented on src/components/WebexAvatar/WebexAvatar.stories.js:
```diff
@@ -1,43 +1,83 @@
+    <WebexAvatar personID={PersonStatus.TYPING} />
+  </WebexData>
+));
+stories.add('wrong PersonID', () => (
```
New line before
```suggestion

stories.add('wrong PersonID', () => (
```

akoushke commented:
> Can you remove `WebexAvatar.scss` if there's nothing in it?
> 
> Also, please have one refactor commit per component changed, and a feat for the provider work (including hook).

I will, after i'm done with addressing all the comments.

akoushke commented:
> Can you remove `WebexAvatar.scss` if there's nothing in it?
> 
> Also, please have one refactor commit per component changed, and a feat for the provider work (including hook).

WebexAvatar.scss has been removed in this PR already

lalli-flores commented on src/components/WebexActivity/WebexActivity.test.js:
```diff
@@ -1,58 +1,49 @@
     test('matches snapshot with "default" props', () => {
       expect(
-        shallow(<Header personID="default" adapter={adapters.peopleAdapter} created={activities.default.created} />)
+        shallow(<Header personID="default" created={jsonData.activities.default.created} adapter={jsonData.people} />)
```
Is `jsonData.people` an adapter?

akoushke replied:
no. it's an object to our unit test.

lalli-flores replied:
I see why this is off, we shouldn't be passing an adapter to `Header` anymore, it need it for the `WebexAvatar`, but `WebexAvatar` has its own adapter now.

akoushke replied:
yeah. I took care of it.

lalli-flores commented on src/components/WebexActivityStream/README.md:
````diff
@@ -19,13 +19,19 @@ To see all the different possible states of the Webex Activity Stream component,
 
    ```js
-   const jsonAdapter = new RoomsJSONAdapter(rooms);
+   const WebexJSONAdapter = {
````
I meant to say, this should be a `WebexJSONAdapter`, as in the object; not that you change the name of the variable.
```suggestion
   const jsonAdapter = new WebexJSONAdapter(jsonData);
```

lalli-flores commented on src/components/WebexAvatar/README.md:
````diff
@@ -19,13 +19,15 @@ To see all the different possible states of the Webex Avatar component, you can
 
    ```js
-   const jsonAdapter = new PeopleJSONAdapter(people);
+   const WebexJSONAdapter = {peopleAdapter: new PeopleJSONAdapter(people)};
````
I meant to say, this should be a `WebexJSONAdapter`, as in the object; not that you change the name of the variable.
```suggestion
   const jsonAdapter = new WebexJSONAdapter(jsonData);
```

akoushke commented on src/components/WebexActivity/WebexActivity.js:
```diff
@@ -34,12 +34,13 @@ export function formatMessageDate(timestamp) {
   return (
     <div className="activity-header">
-      <WebexAvatar personID={personID} adapter={adapter} />
+      <WebexAvatar personID={personID} />
```
this is so nice.

adamweeks commented on src/adapters/WebexJSONAdapter.js:
```diff
@@ -0,0 +1,13 @@
+export default class WebexJSONAdapter {
+  /**
+   * Creates a new instance of the WebexAdapter.
+   * @param {Object} datasource The primary datasource the json adapter will be using.
```
Can we add the required properties?

```suggestion
   * @param {Object} datasource The primary datasource the json adapter will be using.
   * @param {Object} datasource.activities
   * @param {Object} datasource.people
   * @param {Object} datasource.rooms
```

akoushke replied:
I addressed your concern differently. Please take a look and let me know what you think. I prefer to not add an additional param, because that's not what the code does. I did however try to document the structure of our WebexJSONAdapter.

webex-components-publisher commented:
:tada: This PR is included in version 1.16.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@webex/components)
- [GitHub release](https://github.com/webex/components/releases/tag/untagged-b37f5f044bf91f37b51e)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
